engine/runtime-language-epb/src/main/java/org/enso/interpreter/epb/runtime/ForeignParsingException.java

```diff
@@ -0,0 +1,45 @@
+package org.enso.interpreter.epb.runtime;
+
+import com.oracle.truffle.api.exception.AbstractTruffleException;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.library.ExportLibrary;
+import com.oracle.truffle.api.library.ExportMessage;
+import com.oracle.truffle.api.nodes.Node;
+
+/**
+ * This exception is thrown once a foreign method cannot be parsed because the associated foreign
+ * language is not installed, or enabled, in the Truffle engine.
+ */
+@ExportLibrary(InteropLibrary.class)
+public class ForeignParsingException extends AbstractTruffleException {
+  private final String message;
+
+  public ForeignParsingException(String truffleLangId, Node location) {
+    super(createMessage(truffleLangId), location);
+    this.message = createMessage(truffleLangId);
+  }
+
+  private static String createMessage(String truffleLangId) {
+    return "Failed to parse Truffle language with ID " + truffleLangId;
```

Comment:
I would avoid mentioning Truffle in the error message. End users are unlikely to know what Truffle is. Maybe:

"Cannot evaluate foreign " + truffleLangId + ". Only available languages are ...."